web/blog/tenzir-for-splunk-users/index.md

```diff
@@ -0,0 +1,316 @@
+
+### Rare JA3 hashes
+
+Splunk:
+
+```splunk-spl
+index=zeek sourcetype=zeek_ssl
+| rare ja3
+```
+
+Tenzir:
+
+```splunk-spl
+export
+| where #schema == "zeek.ssl"
+| rare ja3
+| head 10
+```
+
+This example shows again how to select a specific data source and perform some
+"stack counting".
+
+### Expired certificates
+
+Splunk:
+
+```splunk-spl
+index=zeek sourcetype=zeek_x509
+| convert num(certificate.not_valid_after) AS cert_expire
+| eval current_time = now(), cert_expire_readable = strftime(cert_expire,"%Y-%m-%dT%H:%M:%S.%Q"), current_time_readable=strftime(current_time,"%Y-%m-%dT%H:%M:%S.%Q")
+| where current_time > cert_expire
+```
+
+Tenzir:
+
+```splunk-spl
+export
+| where certificate.not_valid_after > now()
+```
+
+This example shows the benefit of native time types (and Tenzir's rich type
+system in more generally).
+
+- TQL's [type system](/data-model/type-system) has first-class support for times
+  and durations.
+
+- TQL's [`zeek-tsv`](/formats/zeek-tsv) parser preserves `time` types natively,
+  so you don't have to massage strings at query time.
+
+### Large DNS queries
+
+Splunk:
+
+```splunk-spl
+index=zeek sourcetype=zeek_dns
+| eval query_length = len(query)
+| where query_length > 75
+| table _time id.orig_h id.resp_h proto query query_length answer
+```
+
+Tenzir:
+
+```splunk-spl
+export
+| extend query_length = length(zeek.dns.query)
+| where query_length > 75
+| select #timestamp, id.orig_h, id.resp_h, proto, query, query_length
+```
+
+We don't have functions in TQL. *Yet*. It's one of our most important roadmap
+items at the time of writing, so stay tuned.
+
+If we had functions, we would translate this query as shown above.
+Splunk's `len` would map to Tenzir's `length` function, and the `_time`
+field maps to the `#timestamp` selector.
+
+### Query responses with NXDOMAIN
+
+Splunk:
+
+```splunk-spl
+index=zeek sourcetype=zeek_dns rcode_name=NXDOMAIN
+| table _time id.orig_h id.resp_h proto query
+```
+
+Tenzir:
+
+```splunk-spl
+export
+| where zeek.dns.rcode_name == "NXDOMAIN"
+| select #timestamp, id.orig_h, id.resp_h, proto, query
+```
+
+The `table` operator in splunk outputs the data in tabular form. This is the
+default for our [app](/setup-guides/use-the-app). There's also an
+[upcoming](https://github.com/tenzir/tenzir/pull/3113) `write table` format to
+generate textual representation outside the app.
+
+## Summary
+
+In this blog we've juxtaposed the languages of Splunk (SPL) and Tenzir (TQL).
+They are remarkably similar—and that's not accidental. When we talked to
+security analysts we often heard that Splunk has a great UX. Even our own
+engineers that live on the command line find this mindset natural. But Splunk
+was not our only inspiration, we also drew inspiration from Kusto and others.
+
+As we created TQL, we wanted to learn from missed opportunities and
+bolt-on growth from past efforts, while at the same time doubling down on the
+successes others have confirmed.
```

Comment:
"learn from bolt-on growth from past efforts" and "successes others have confirmed" sound not very natural to me.